app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt

```diff
@@ -88,32 +95,24 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
             // generated code.
             register(observer)
 
-            // This performs the minimal amount of work required to load branch and enrolment data
-            // into memory. If `getExperimentBranch` is called from another thread between here
-            // and the next nimbus disk write (setting `globalUserParticipation` or
-            // `applyPendingExperiments()`) then this has you covered.
-            // This call does its work on the db thread.
-            initialize()
+            val isFirstNimbusRun = context.settings().isFirstNimbusRun || url.isNullOrBlank()
```

Comment:
I'm not sure if this make sense to have the url check as part of the first run.

```suggestion
            val isFirstNimbusRun = context.settings().isFirstNimbusRun
```

Semantically, I think it's ok, but it may not always. I've added the url check below.